### DISCUSSION TITLE
Select All Check box with controlled inputs

### DISCUSSION THREAD

**Discussion by nageshwari17** [react-hook-form/react-hook-form#7729] [Q&A] [answered]
Hi , 

I am working on filters in my project . there in need to have select all filter in particular sections . If I check select all, all value should select w.r.t section and if you uncheck one of the value in particular section I need to uncheck select all option .  Please have a look into sandbox examples .

[https://codesandbox.io/s/little-sunset-5s3j2?file=/src/App.tsx](https://codesandbox.io/s/little-sunset-5s3j2?file=/src/App.tsx).

Please let me know how do I implement this feature .

**Desktop (please complete the following information):**

OS: window
Browser : Chrome
Version: Window 10

**Comment by Moshyfawn** [comment, marked as answer]
Take a look at [this CSB](https://codesandbox.io/s/keen-fire-tf676?file=/src/Demo.tsx). Might need a bit of a cleanup, but the main logic is there

**Reply by nageshwari17** [reply]
Thanks for your Quick replay @Moshyfawn :).

Do you have any idea how to map nested check box (in tree structure)?

**Reply by Moshyfawn** [reply]
What do you mean by "map nested check box"?
If you want to dynamically show another set of checkboxes if some condition for the previous checkbox group met, you can [`watch`](https://react-hook-form.com/api/useform/watch) for the value change and conditionally render it.
```js
const firstCheckboxGroup = watch("firstCheckboxGroup")

if (firstCheckboxGroup.some(value => value === "all")) {
  return (
    <AnotherCheckboxGroup />
  )
}
```
More of a pseudo-code than an actual code, but here's the main gist of it

**Reply by nageshwari17** [reply]
Its some like 
![image](https://user-images.githubusercontent.com/61220343/152334550-425fca0f-b0bf-484b-95a8-d92535820cc9.png)
